graylog2-server/src/main/java/org/graylog2/contentpacks/facades/GrokPatternFacade.java

```diff
@@ -166,4 +173,69 @@ public EntityExcerpt createExcerpt(GrokPattern grokPattern) {
             return Optional.empty();
         }
     }
+
+    @Override
+    public Graph<EntityDescriptor> resolveNativeEntity(EntityDescriptor entityDescriptor) {
+        final MutableGraph<EntityDescriptor> mutableGraph = GraphBuilder.directed().build();
+        mutableGraph.addNode(entityDescriptor);
+
+        final ModelId modelId = entityDescriptor.id();
+        try {
+            final GrokPattern grokPattern = grokPatternService.load(modelId.id());
+
+            final Set<String> namedGroups = GrokUtils.getNameGroups(GrokUtils.GROK_PATTERN.pattern());
+            final String namedPattern = grokPattern.pattern();
+            final Matcher matcher = GrokUtils.GROK_PATTERN.matcher(namedPattern);
+            while (matcher.find()) {
+                final Map<String, String> group = GrokUtils.namedGroups(matcher, namedGroups);
+                final String patternName = group.get("pattern");
+                grokPatternService.loadByName(patternName).ifPresent(depPattern -> {
+                    final EntityDescriptor depEntityDescriptor = EntityDescriptor.create(
+                            depPattern.id(), ModelTypes.GROK_PATTERN_V1);
+                    mutableGraph.putEdge(entityDescriptor, depEntityDescriptor);
+                });
+            }
+        } catch (NotFoundException e) {
+            LOG.debug("Couldn't find grok pattern {}", entityDescriptor, e);
+        }
+        return mutableGraph;
+    }
+
+    @Override
+    public Graph<Entity> resolveForInstallation(Entity entity,
+                                                Map<String, ValueReference> parameters,
+                                                Map<EntityDescriptor, Entity> entities) {
+        if (entity instanceof EntityV1) {
+            return resolveForInstallationV1((EntityV1) entity, parameters, entities);
+        } else {
+            throw new IllegalArgumentException("Unsupported entity version: " + entity.getClass());
+        }
+    }
+
+    private Graph<Entity> resolveForInstallationV1(EntityV1 entity,
+                                                   Map<String, ValueReference> parameters,
```

Comment:
The `parameters` are currently not used. We need to use them to resolve parameterized grok patterns. A user could parameterize the grok pattern iteself. If that happens we need to resolve the parameter. For example:

```java
final String namedPattern = grokPatternEntity.pattern().asString(parameters);
```

I doesn't really make sense to parameterize that kind of data, but I think we still have to do it correctly. :wink: